src/components/LHNOptionsList/BaseOptionsListLHN.js

```diff
@@ -0,0 +1,136 @@
+import _ from 'underscore';
+import React, {forwardRef, Component} from 'react';
+import {View, FlatList} from 'react-native';
+import PropTypes from 'prop-types';
+import styles from '../../styles/styles';
+import OptionRowLHN from './OptionRowLHN';
+import variables from '../../styles/variables';
+import {propTypes as optionsListPropTypes, defaultProps as optionsListDefaultPropTypes} from './optionsListPropTypesLHN';
+import CONST from '../../CONST';
+
+const propTypes = {
+    /** Determines whether the keyboard gets dismissed in response to a drag */
+    keyboardDismissMode: PropTypes.string,
+
+    /** Called when the user begins to drag the scroll view */
+    onScrollBeginDrag: PropTypes.func,
+
+    ...optionsListPropTypes,
+};
+
+const defaultProps = {
+    keyboardDismissMode: 'none',
+    onScrollBeginDrag: () => {},
+    ...optionsListDefaultPropTypes,
+};
+
+class BaseOptionsListLHN extends Component {
+    constructor(props) {
+        super(props);
+
+        this.renderItem = this.renderItem.bind(this);
+        this.getItemLayout = this.getItemLayout.bind(this);
+        this.onViewableItemsChanged = this.onViewableItemsChanged.bind(this);
+        this.viewabilityConfig = {viewAreaCoveragePercentThreshold: 95};
+        this.didLayout = false;
+    }
+
+    shouldComponentUpdate(nextProps) {
+        if (nextProps.focusedIndex !== this.props.focusedIndex) {
+            return true;
+        }
+
+        if (nextProps.headerMessage !== this.props.headerMessage) {
+            return true;
+        }
+
+        if (!_.isEqual(nextProps.sections, this.props.data)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    onViewableItemsChanged() {
+        if (this.didLayout || !this.props.onLayout) {
+            return;
+        }
+
+        this.didLayout = true;
+        this.props.onLayout();
+    }
+
+    /**
+     * This function is used to compute the layout of any given item in our list.
+     * We need to implement it so that we can programmatically scroll to items outside the virtual render window of the SectionList.
```

Comment:
I love how you're asking questions on a bunch of old code that was just copy/pasted 😁 

Here was the original PR:https://github.com/Expensify/App/pull/7702

The explanation for this was:
> When a single item is selected, arrow navigation does not always scroll to the right position (the selected item is sometimes partially or fully covered)

This is great to find because this is something else that can be cleaned up since no selection stuff is happening and there is no arrow navigation. I'll clean this up.

Reply:
haha yeah it seems like it is very "useful somewhere", but not here (which sort of implies it was added in the wrong spot)